dashboard/lib/metrics/events.rb

```diff
@@ -46,11 +46,21 @@ def log_event(user: nil, event_name:, event_value: nil, metadata: {}, get_enable
       )
       end
 
+      def log_event_with_session(session:, event_name:, event_value: nil, metadata: {})
```

Comment:
Thinking through this, I'm wondering if this would be cleaner if we make this a private method, and conditionally call this or the `log_statsig_event_with_cdo_user` in the main `log_event` method depending on if `user` is nil?

Reply:
I wrote up some of the pros/cons I can think of, I'm leaning toward Option 1 to avoid possible bugs if someone forgets to input `user` and the analytics get chained incorrectly.

### Option 1: Create a `log_event_with_session` method, the caller must use this if logging with user.

Pros: 
* Explicitly known that the session is used to store
* Oddball compared with `log_event`, so little more clear that this is special

Cons:
* Extra method, might be a bit confusing

### Option 2: Add `session` field to the parameter

Pros:
* Less methods to consider, always use `log_event` to log

Cons:
* If the caller accidentally forgets `user`, it will log with the stable id resulting in possible bugs where the metrics aren't chained correctly

Reply:
If we go with option 1, I think this new method needs to handle the same cases as the `log_event` method, namely:

* set event_value to event_name if nil
* log_event_to_stdout if env == :development
* It should probably have exception handling too?

Reply:
I think I'm comfortable with option 2, where we default to the stable id if user is nil. This should be easy to catch cases where `user` is accidentally left out, and can be corrected. I think the simplicity of this option outweighs the potential downsides.

Reply:
Updated to use the same logic as above (might be some opportunity to clean up after we get Statsig working for anonymous users)

```
Logging Event: {"user_id":"1a02b095-9308-46ac-b1c1-7e254c74d333","event_name":"clever-begin-auth","event_value":"clever-begin-auth","metadata":{}}
Logging Event: {"user_id":"1a02b095-9308-46ac-b1c1-7e254c74d333","event_name":"clever-callback","event_value":"clever-callback","metadata":{"study":"account-sign-up-v5","study_group":"not-in-study"}}
```